File: go/secret-handshake/secret_handshake.go
package secret

const (
	Wink uint = 1 << iota
	DoubleBlink
	CloseEyes
	Jump
	Reverse
)

var actions = []struct {
	mask uint
	text string
}{
	{Wink, "wink"},
	{DoubleBlink, "double blink"},
	{CloseEyes, "close your eyes"},
	{Jump, "jump"},
}

func Handshake(code uint) []string {
	var result []string
	rev := code&Reverse != 0
	for i := 0; i < len(actions); i++ {
		a := actions[i]
		if rev {
			a = actions[(len(actions)-1)-i]
		}
		if a.mask&code != 0 {
			result = append(result, a.text)
		}
	}
	return result
}

File: c/secret-handshake/secret_handshake.c
#include "secret_handshake.h"

#include <stdlib.h>
#include <stdint.h>
#include <string.h>

#define CHECK_BIT(bitmap, idx) ((bitmap & (1 << idx)) == (1 << idx))

static void append(char** arr, char* str, size_t pos) {
   arr[pos] = malloc(15*sizeof(char));
   strcpy(arr[pos], str);
}
const char **commands(size_t number) {
   char **res = calloc(4, sizeof(char*));
   size_t pos = 0;
   if (CHECK_BIT(number, 4)) {
      if (CHECK_BIT(number, 3)) append(res, "jump", pos++);
      if (CHECK_BIT(number, 2)) append(res, "close your eyes", pos++);
      if (CHECK_BIT(number, 1)) append(res, "double blink", pos++);
      if (CHECK_BIT(number, 0)) append(res, "wink", pos++);
   } else {
      if (CHECK_BIT(number, 0)) append(res, "wink", pos++);
      if (CHECK_BIT(number, 1)) append(res, "double blink", pos++);
      if (CHECK_BIT(number, 2)) append(res, "close your eyes", pos++);
      if (CHECK_BIT(number, 3)) append(res, "jump", pos++);
   }
   return (const char**) res;
}
